internal/kafka/command_topic_produce_onprem.go

```diff
@@ -171,15 +171,20 @@ func (c *command) produceOnPrem(cmd *cobra.Command, args []string) error {
 	signal.Notify(signals, os.Interrupt)
 	go func() {
 		<-signals
-		close(input)
+		input <- EOF
 	}()
 	go scan() // Prime reader
 
 	deliveryChan := make(chan ckafka.Event)
 	for data := range input {
 		if data == "" {
+			if scanErr != nil {
+				break
+			}
 			go scan()
 			continue
+		} else if data == EOF {
```

Comment:
I wonder if there's a good way to combine the cloud/on-prem implementations? It's annoying that we have to keep updating both.

Reply:
Yeah, I think that would be good, but it might be better to put in another PR, since there's other `kafka` and even `kafka topic` commands that whose cloud/on-prem implementations share a lot of code, and that would end up being a lot of code refactoring.